Aptoide/aptoide-client-v8
# [MOB-478] Fixed installing state when app is in background and android 10

**What does this PR do?**

This PR aims at fixing the installing state in Android 10. We added a new state so that we can show the install button before popping the install dialog. This is important because if the app is in background, we won't be able to show the install dialog and then when the user comes back, he will see the appview of that app in the "installing state" (as we can not hear the error of not popping the dialog). 
Now, since we show the install button before popping the dialog, we won't have this problem.
So now we should show the install button before popping the dialog and then set to indeterminate when we get the callback that the user pressed the install button -> This was discovered by hearing the progress of the installation of all apps, noting that it always sets the progress to a certain number after the user presses the install button.

**Database changed?**

   Yes

**Where should the reviewer start?**

- [ ] AppInstaller.java

**How should this be manually tested?**

  On Android 10 or above, try to download an app, set the app to background. Note that when you go the appview it is not showing indeterminate. Also, test the flow where the app is alive, check if it is coherent.
Also test on other android versions below 10 and test on xiaomi device.

**What are the relevant tickets?**

  Tickets related to this pull-request: [MOB-478](https://aptoide.atlassian.net/browse/MOB-478)

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new keys, etc.) 



**Code Review Checklist**
